crates/query-engine/translation/src/translation/query/native_queries.rs

```diff
@@ -72,5 +75,38 @@ pub fn translate(env: &Env, state: State) -> Result<Vec<sql::ast::CommonTableExp
         });
     }
 
-    Ok(ctes)
+    Ok((ctes, global_table_index))
```

Comment:
We return the global table index here as well so it can be used when wrapping ctes, but the native queries with the state is consumed.